### DISCUSSION TITLE
Entities not availabe

### DISCUSSION THREAD

**Discussion by BFResende** [tsightler/ring-mqtt#804] [Support Q&A] [answered]
Hi!

So I've been using this integration for a couple of months and with this last home assistant update all my entites became unavailable.

I've already reconfigured everything and nothing changes.

Any recomendation to solve this problem?

<img width="1423" alt="Screenshot 2024-03-17 at 22 12 03" src="https://github.com/tsightler/ring-mqtt/assets/41636150/0bc72349-3e60-45ef-bb1e-f799dfe8ad0a">
<img width="1093" alt="Screenshot 2024-03-20 at 11 44 51" src="https://github.com/tsightler/ring-mqtt/assets/41636150/fb2b27f3-1aee-4a7e-8046-d40407c21122">
<img width="1283" alt="Screenshot 2024-03-20 at 11 45 03" src="https://github.com/tsightler/ring-mqtt/assets/41636150/3da0e918-a872-4d50-9bad-5279e7be9a06">

**Comment by tsightler** [comment]
https://github.com/tsightler/ring-mqtt/wiki/Support-&-Troubleshooting#troubleshooting

**Comment by BFResende** [comment]
I have already done everything.

The integration catches again all the devices and entities but they stay unavailble.

**Reply by tsightler** [reply]
No idea what "cathes again all the devices" means, but if the logs show that ring-mqtt is connected and sending updates, then there's nothing more to be done on the ring-mqtt side and you'll have to troubleshoot elsewhere.  You just provided screenshots of devices/entities showing offline which is roughly as useful as taking pictures of a car broken down on the side of the road and asking what is wrong with it.

**Reply by BFResende** [reply]
I'm sorry, but I'm not an expert.

I presume that the logs are the photo bellow right?

<img width="535" alt="Screenshot 2024-03-20 at 14 49 49" src="https://github.com/tsightler/ring-mqtt/assets/41636150/57632c24-4b25-4a45-8784-7a0c1965f191">

**Reply by tsightler** [reply]
The screenshot clearly shows that discovery is not completing.  Please read the message that says 'IMPORTANT' in the log, also covered in step 2 of the troubleshooting steps I linked earlier.  It seems there is some offline device at "Casa Amadora" as that is where device discovery is hanging.

**Comment by BFResende** [comment]
Also sending the Mosquitto Broker logs...

<img width="1083" alt="Screenshot 2024-03-20 at 14 57 49" src="https://github.com/tsightler/ring-mqtt/assets/41636150/39d6c8d9-2b4e-4251-8ed2-31bf0cf13c3c">

**Comment by BFResende** [comment]
Thank you very much for your help but all my devices are online at least in CASA AMADORA.

**Reply by tsightler** [reply]
In over 5 years I have never seen a case where device discovery hangs that did not involve an offline device, or at least a hub that is in failover state (i.e. using cellular backup, which still counts as offline).  There have been a small number of cases where a hub needed to be rebooted even though it didn't show fully offline in the Ring app, but it was still technically offline.

I've had people argue with me, curse at me, call me names, only to take my personal time and immediately find a smart lighting bridge they never removed from the app, or a light bulb that is offline or whatever.  I'm not saying 100% that it couldn't be something else, but as I've never seen a case with 10's of thousands of users across ring-mqtt and other projects that use ring-client-api, I no longer spend my personal time to help people find their offline device.

I would suggest temporarily restricting which sites your addon uses using the locations feature and see if discovery continues for the other sites.  Then you can work on figuring out what is blocking discovery in the offending site.

**Reply by BFResende** [reply, marked as answer]
Sure I totally understand. I do appreciate the time o spend trying to help. 

In the end I manage to solve the problem. The solution was to remove everything from my ring app and add it again.

Don't know what happened but all my devices were online at the time.

It's solved thank you very much.

**Comment by BFResende** [comment]
Well it seems that the problem is not solved...

I manage to put everthing running well but after 5 minutes the FRONT ENTRANCE (INTERCOM) just goes offline.

Then I just STOP the addon and START again e comes back online an then again after 5 minutes becames offiline again.

I'm in a loop! 

<img width="728" alt="Screenshot 2024-03-22 at 12 32 14" src="https://github.com/tsightler/ring-mqtt/assets/41636150/97765b2b-b13c-403e-82e4-fc15a11b36b3">

<img width="765" alt="Screenshot 2024-03-22 at 12 32 26" src="https://github.com/tsightler/ring-mqtt/assets/41636150/59d64155-2526-4b53-81cd-1501840e9253">

<img width="738" alt="Screenshot 2024-03-22 at 12 32 54" src="https://github.com/tsightler/ring-mqtt/assets/41636150/79797853-4198-4622-b7ab-ec4789883f74">

<img width="763" alt="Screenshot 2024-03-22 at 12 33 03" src="https://github.com/tsightler/ring-mqtt/assets/41636150/2f94dbe9-5031-4715-b97e-1b72d2d9eb05">

**Reply by BFResende** [reply]
<img width="722" alt="Screenshot 2024-03-22 at 16 03 15" src="https://github.com/tsightler/ring-mqtt/assets/41636150/ab8b7f49-f2c7-4fdf-aabc-ec1eada059ce">

**Reply by BFResende** [reply]
This is whats keeps happening...

**Reply by tsightler** [reply]
The code is only reporting what it happening, the websocket is being closed.  This would indicate some type of networking issue.  The fact that you also mention that the intercom goes offline, which uses a API polling method vs a websocket, also indicates the same.  Unfortunately, as this isn't a code related issue, I don't really have any way to help you further.  I would check things like making sure the network is 100% reliable, making sure you have a local caching DNS (very important for ring-client-api to work correctly when there are a lot of devices), etc.

**Comment by BFResende** [comment]
Well I’m not being able to solve this.

In the ring app all the devices are always online and they never go offline but in the ring MQTT it seems they are.

Already tried everything…
